ml-dhs-aws/src/main/java/com/marklogic/client/dhs/JobRunner.java

```diff
@@ -64,22 +87,64 @@ public void run(DatabaseClient client, InputStream csvRecords, InputStream jobCo
                         throwable.printStackTrace(System.out);
                      });
 
-         ObjectMapper objectMapper = new ObjectMapper();
-         ObjectNode jobControlObj = (ObjectNode) objectMapper.readTree(jobControl);
-         ObjectNode jobDef        = (ObjectNode) jobControlObj.get("job");
-         ObjectNode recordDef     = (ObjectNode) jobControlObj.get("record");
 
          ObjectLoader loader = new ObjectLoader(batcher, recordDef, getJobDirectory(), documentMetadata);
-         CSVConverter converter = new CSVConverter();
-
-// TODO: write before job document in /jobStart and /jobs/ID collections or send before job payload to DHF endpoint
+         CSVConverter converter = new CSVConverter();  
 
+         Iterator<ObjectNode> itr = converter.convertObject(csvRecords);
+         if(!itr.hasNext()) {
+        	 throw new MarkLogicIOException("No header found.");
+         }
+         ObjectNode csvNode = itr.next();
+         
+         String headerValue = csvNode.fieldNames().toString();
+         
+   // write before job document in /jobStart and /jobs/ID collections or send before job payload to DHF endpoint        
+         String beforeDocId = "/jobs/"+id+"/beforeJob.json";
+         
+         DocumentMetadataHandle beforeDocumentMetadata = new DocumentMetadataHandle();
+         beforeDocumentMetadata.withCollections("/jobs/"+beforeDocId, "/beforeJob");
+         
+         String ingestionStartTime = LocalDateTime.now().toString();
+         
+         ObjectNode beforeDocRoot = objectMapper.createObjectNode();
+         beforeDocRoot.put("job id", id);
+         beforeDocRoot.put("jobMetadataValue",  jobMetadataValue);
+         beforeDocRoot.put("ingestionStartTime",  ingestionStartTime);
+         beforeDocRoot.put("headerValue",  headerValue);
+         
+         JacksonHandle jacksonHandle = new JacksonHandle(beforeDocRoot);
+         jsonMgr.write(beforeDocId, beforeDocumentMetadata, jacksonHandle);
+         
          JobTicket ticket = moveMgr.startJob(batcher);
-         loader.loadRecords(converter.convertObject(csvRecords));
+         loader.loadRecord(csvNode);
+         
+         loader.loadRecords(itr);
          batcher.flushAndWait();
          moveMgr.stopJob(ticket);
+         
+         String ingestionStopTime = LocalDateTime.now().toString();
+
+// write after job document with job metadata in /jobEnd and /jobs/ID collections or send after job payload to DHF endpoint
+         String afterDocId = "/jobs/"+id+"/afterJob.json";
+         
+         DocumentMetadataHandle afterDocumentMetadata = new DocumentMetadataHandle();
+         afterDocumentMetadata.withCollections("/jobs/"+afterDocId, "/afterJob");
+         
+         ObjectNode afterDocRoot = objectMapper.createObjectNode();
+         afterDocRoot.put("job id", id);
+         afterDocRoot.put("jobMetadataValue",  jobMetadataValue);
```

Comment:
As with the before document, does it make sense to name the key "jobMetadata" and to make the value the unserialized metadataNode?